components/permissions/contexts/brave_ethereum_permission_context.cc

```diff
@@ -227,6 +228,35 @@ void BraveEthereumPermissionContext::GetAllowedAccounts(
   std::move(callback).Run(true, allowed_accounts);
 }
 
+// static
+bool BraveEthereumPermissionContext::AddEthereumPermission(
+    content::BrowserContext* context,
+    const std::string& origin_spec,
+    const std::string& account) {
+  bool has_permission;
+  if (!HasEthereumPermission(context, origin_spec, account, &has_permission))
+    return false;
+
+  // If the user already has permission, then return false
+  // since the call didn't change anything.
+  if (has_permission)
+    return false;
```

Comment:
I went back and forth in my head here. I don't think it matters really because if you want to know if it has permission you can use HasEthereumPermission.  I guess I'll update it so it will be idempotent which is a nice property.

Reply:
Fixed in rebase and added a test case for it